cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -862,6 +866,94 @@ instance FromJSON DelegationsAndRewards where
        rewardAccountBalance <- o .:? "rewardAccountBalance"
        pure (address, rewardAccountBalance, delegation)
 
+runQueryLeadershipSchedule
+  :: AnyConsensusModeParams
+  -> NetworkId
+  -> GenesisFile -- ^ Shelley genesis
+  -> Hash StakePoolKey
+  -> SigningKeyFile -- ^ VRF signing key
+  -> ExceptT ShelleyQueryCmdError IO ()
+runQueryLeadershipSchedule (AnyConsensusModeParams cModeParams) network
+                           (GenesisFile genFile) poolid (SigningKeyFile vrfSkeyFp) = do
+  SocketPath sockPath <- firstExceptT ShelleyQueryCmdEnvVarSocketErr readEnvSocketPath
+  let localNodeConnInfo = LocalNodeConnectInfo cModeParams network sockPath
+
+  anyE@(AnyCardanoEra era) <- determineEra cModeParams localNodeConnInfo
+  sbe <- getSbe $ cardanoEraStyle era
+  let cMode = consensusModeOnly cModeParams
+  vrkSkey <-firstExceptT ShelleyQueryCmdTextEnvelopeReadError . newExceptT
+               $ readFileTextEnvelope (AsSigningKey AsVrfKey) vrfSkeyFp
+  shelleyGenesis <- firstExceptT ShelleyQueryCmdGenesisReadError $
+                          newExceptT $ readAndDecodeShelleyGenesis genFile
+  case cMode of
+    CardanoMode -> do
+      eInMode <- toEraInMode era cMode
+          & hoistMaybe (ShelleyQueryCmdEraConsensusModeMismatch (AnyConsensusMode cMode) anyE)
+
+      let pparamsQuery = QueryInEra eInMode $ QueryInShelleyBasedEra sbe QueryProtocolParameters
+          serDebugLedgerStateQuery = QueryInEra eInMode . QueryInShelleyBasedEra sbe $ QueryDebugLedgerState
```

Comment:
What's the `ser` prefix?  Does it mean Shelly ERa?

Reply:
Serialised